src/ui/MultipleFilesMessageItemBody/useDynamicUploadedIndex.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import { MultipleFilesMessage, SendingStatus } from '@sendbird/chat/message';
+import { createStatefulFileInfoList, StatefulFileInfo } from '../../utils/createStatefulFileInfoList';
+
+export const useDynamicUploadedIndex = (message: MultipleFilesMessage): StatefulFileInfo[] => {
+  const [statefulFileInfoList, setStatefulFileInfoList] = useState<StatefulFileInfo[]>(
+    message['statefulFileInfoList'] ?? message.fileInfoList,
+  );
+
+  useEffect(() => {
+    if (
+      message?.sendingStatus !== SendingStatus.SUCCEEDED
+      && Array.isArray(message['statefulFileInfoList'])
+    ) {
+      const newStatefulFileInfoList: StatefulFileInfo[] = createStatefulFileInfoList(message, statefulFileInfoList);
+      setStatefulFileInfoList(newStatefulFileInfoList);
+    }
+  }, [message['statefulFileInfoList']?.map((fileInfo: StatefulFileInfo) => fileInfo.isUploaded).join(',')]);
```

Comment:
It is expected that the timing of changes to the `message.statefulFileInfoList` aligns with the timing of changes to each `fileInfo.isUploaded`.

Therefore, it seems that making the array to string using the .`join` in the code is unnecessary.